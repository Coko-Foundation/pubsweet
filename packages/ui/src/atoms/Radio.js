import React from 'react'
import styled from 'styled-components'
import { th, override } from '@pubsweet/ui-toolkit'

const Label = styled.span`
  ${override('ui.Radio.Label')};
`
const Input = styled.input`
  margin-right: ${th('subGridUnit')};
  ${override('ui.Radio.Input')};
`
const Root = styled.label`
  cursor: pointer;
  align-items: center;
  display: ${props => (props.inline ? 'inline-flex' : 'flex')};
  min-height: calc(${th('gridUnit')} * 3);
  color: ${props => (props.color ? props.color : props.theme.colorText)};

  ${override('ui.Radio')};
`
const Radio = ({
  className,
  color,
  inline,
  name,
  value,
  label,
  checked,
  required,
  onChange,
  readonly,
}) => (
  <Root checked={checked} color={color} inline={inline}>
    <Input
      checked={checked}
      disabled={readonly}
      name={name}
      onChange={onChange}
      required={required}
      type="radio"
      value={value}
    />
    <Label checked={checked} color={color}>
      {label}
    </Label>
  </Root>
)

export default Radio
