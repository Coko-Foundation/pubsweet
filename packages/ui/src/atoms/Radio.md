A radio button.

```js
import { useState } from 'react'
const [state, setState] = useState({ value: undefined })

;<Radio
  name="radio"
  checked={state.value === 'on'}
  onChange={event => setState({ value: event.target.value })}
/>
```

A checked radio button.

```js
import { useState } from 'react'
const [state, setState] = useState({ value: 'on' })

;<Radio
  name="radio-checked"
  checked={state.value === 'on'}
  onChange={event => setState({ value: event.target.value })}
/>
```

A radio button with a label.

```js
import { useState } from 'react'
const [state, setState] = useState({ value: undefined })

;<Radio
  name="radio-checked"
  label="Foo"
  checked={state.value === 'on'}
  onChange={event => setState({ value: event.target.value })}
/>
```

A radio button with a color.

```js
import { useState } from 'react'
const [state, setState] = useState({ value: undefined })

;<Radio
  name="radio-color"
  label="Foo"
  color="red"
  checked={state.value === 'on'}
  onChange={event => setState({ value: event.target.value })}
/>
```
