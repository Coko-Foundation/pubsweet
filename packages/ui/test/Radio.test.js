import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import Radio from '../src/atoms/Radio'
import TestThemeProvider from './setup/theme'

const props = {
  checked: false,
  label: 'TestLabel',
  name: 'TestName',
  required: true,
  value: 'TestValue',
}

describe('Radio', () => {
  test('is rendered correctly', () => {
    const tree = renderer
      .create(
        <TestThemeProvider>
          <Radio {...props} />
        </TestThemeProvider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Input gets the correct props', () => {
    const input = mount(<Radio {...props} />).find('input')

    expect(input.prop('name')).toBe(props.name)
    expect(input.prop('value')).toBe(props.value)
    expect(input.prop('checked')).toBe(props.checked)
    expect(input.prop('required')).toBe(props.required)
  })
})
