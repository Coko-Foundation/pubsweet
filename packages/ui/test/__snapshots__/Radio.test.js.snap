// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio is rendered correctly 1`] = `
.c2 {
  position: absolute;
  opacity: 0;
  z-index: -1;
}

.c4 {
  display: inline-block;
  content: ' ';
  width: calc(8px * 2);
  height: calc(8px * 2);
  vertical-align: center;
  margin-left: 8px;
  margin-right: 8px;
  border: calc(8px / 4) solid white;
  border-radius: 50%;
  -webkit-transition: border 1s ease;
  transition: border 1s ease;
  color: black;
}

.c6 {
  display: inline-block;
  font-family: inherit;
  font-size: 18px;
  font-style: italic;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-transition: all 1s;
  transition: all 1s;
  min-height: 32px;
  color: black;
}

.c0:not(:last-child) {
  margin-right: 0;
  margin-bottom: 0;
}

.c0 .c3 {
  background: transparent;
  box-shadow: 0 0 0 1px currentcolor;
}

.c0:hover .c5 {
  color: black;
}

.c0:hover .c3 {
  -webkit-animation-name: kLnLOV;
  animation-name: kLnLOV;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  box-shadow: 0 0 0 1px black;
}

.c0 .c1:focus + .c3 {
  box-shadow: 0 0 1px calc(1px * 2) black;
}

<label
  checked={false}
  className="c0"
  color="black"
>
  <input
    checked={false}
    className="c1 c2"
    disabled={undefined}
    name="TestName"
    onChange={undefined}
    required={true}
    type="radio"
    value="TestValue"
  />
   
  <span
    className="c3 c4"
    color="black"
  />
   
  <span
    className="c5 c6"
  >
     
    TestLabel
     
  </span>
   
</label>
`;
